umcontroller/umcontroller.go

```diff
@@ -1021,6 +1041,51 @@ func (umCtrl *Controller) updateComplete(ctx context.Context, e *fsm.Event) {
 	umCtrl.cleanupCurrentComponentStatus()
 }
 
+func (umCtrl *Controller) updateNodeInfo(nodeInfoProvider NodeInfoProvider) error {
+	ids, err := nodeInfoProvider.GetAllNodeIDs()
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	for _, id := range ids {
+		nodeInfo, err := nodeInfoProvider.GetNodeInfo(id)
+		if err != nil {
+			return aoserrors.Wrap(err)
+		}
+
+		if nodeInfo.Status == "unprovisioned" {
+			continue
+		}
+
+		umCtrl.connections = append(umCtrl.connections, umConnection{
+			umID:          nodeInfo.NodeID,
+			isLocalClient: nodeInfo.NodeID == umCtrl.currentNodeId, updatePriority: 1000, handler: nil,
+		})
+	}
+
+	return nil
+}
+
+func (umCtrl *Controller) handleNodeInfoChange(nodeInfo *cloudprotocol.NodeInfo) {
+	if nodeInfo.Status == "unprovisioned" {
+		// TODO: node should disconnect itself if it changed status to unprovisioned, right?
```

Comment:
It should, but may not :) And if node behaves inappropriately we should drop it I guess.
Remove TODO

Reply:
removed TODO comment